spec/unit/shoulda/matchers/active_model/validate_confirmation_of_matcher_spec.rb

```diff
@@ -60,4 +60,50 @@
         to validate_confirmation_of(builder.attribute_to_confirm)
     end
   end
+
+  context 'when the writer method for the attribute changes incoming values' do
+    context 'and the matcher knows nothing of this' do
+      it 'raises a CouldNotSetAttributeError' do
+        builder = builder_for_record_validating_confirmation
+
+        builder.model.class_eval do
+          def password=(value)
+            super(value.upcase)
+          end
+        end
+
+        assertion = lambda do
+          expect(builder.record).to validate_confirmation_of(:password)
+        end
+
+        expect(&assertion).to raise_error(
+          Shoulda::Matchers::ActiveModel::AllowValueMatcher::CouldNotSetAttributeError
```

Comment:
Line is too long. [86/80]<br>Put a comma after the last parameter of a multiline method call.